### PR TITLE
Upgrade compliance checker core to 3.1.1 from pypi

### PR DESCRIPTION
**Pull request by anguss00**
Together with https://github.com/aodn/ci-config/pull/152 and https://github.com/aodn/python-aodncore/pull/43

### CODE REVIEW

File: cookbooks/imos_po/recipes/pipeline.rb [hunk 1]
```diff
@@ -204,7 +203,7 @@
     cwd node['imos_po']['pipeline']['local_aodndata_repo']
     virtualenv environment_dir
   end
-else
+ else
```
Comment (mhidas) [inline cookbooks/imos_po/recipes/pipeline.rb]: Extra space unintended?

**Comment by mhidas** [comment]
Does this apply to v1 pipelines as well? It should be only for v2.

**Comment by anguss00** [comment]
Yes, this will only install compliance checker 3.1.1 in the venv for v2 pipeline

File: cookbooks/imos_po/recipes/pipeline.rb [hunk 2]
```diff
@@ -171,10 +171,10 @@
   virtualenv environment_dir
 end
 
-cc_artifact_path = Chef::Recipe::PipelineUtils.get_deployed_artifact_path(node['imos_po']['pipeline']['compliance_checker_artifact'], node)
-python_execute 'install_compliance-checker' do
-  command "-m pip install #{cc_artifact_path}"
+python_package 'compliance-checker' do
+  version node['imos_po']['pipeline']['compliance_checker_version']
   virtualenv environment_dir
+  action :upgrade
```
Comment (lwgordonimos) [inline cookbooks/imos_po/recipes/pipeline.rb]: My only comment is around this bit. It looks correct, but my level of confidence in the poise behaviour is not overly high. Wondering how we can test this in all the relevant scenarios?

e.g.
* no version installed
* lower version installed
* same version installed
* higher version installed

Comment (anguss00) [reply cookbooks/imos_po/recipes/pipeline.rb]: Yep, definitely. I'll test this out with all of those cases.

Comment (anguss00) [reply cookbooks/imos_po/recipes/pipeline.rb]: happy

**Comment by anguss00** [comment]
Ready to go @mhidas @lwgordonimos

**Pull request merged**